### PR TITLE
add page on beta releases & release candidates for EasyBuild v5.0

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Review by branfosj — APPROVED** [review]

File: docs/easybuild-v5/release-candidates.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+- using **Lmod >= v8.0** or **Environment Modules v4.3.0** is required as modules tool (check with `module --version`);
+- a new function named `run_shell_cmd` is used to run shell commands like `cmake`, `make`, `pip`, etc. (more info [here](run_shell_cmd.md));
+- the **default settting of some configuration options has changed**:
+    - **RPATH linking** is used by default (can be disabled with `--disable-rpath`);
+    - **trace output** is enabled by default (can be disabled with `--disable-trace`);
+    - Lua module files generated by EasyBuild will use `depends_on` rather than `load` by default (can be disabled with `--disable-module-depends-on`);
+    - the `use_pip` and `sanity_pip_check` custom easyconfig parameters used by the `PythonPackage` and `PythonBundle` easyblocks are enabled by default;
+- **some functionality has been deprecated**, including the use of the `run_cmd` and `run_cmd_qa` functions (`run_shell_cmd` should be used instead);
+- easyconfigs using an old [unsupported toolchain](../policies/toolchains.md) have been moved to the [easyconfigs archive](https://github.com/easybuilders/easybuild-easyconfigs-archive);
+
```
Comment (lexming) [inline docs/easybuild-v5/release-candidates.md:32]: Maybe also mention:
* EB_PYTHON_PREFIX
* reproducible tarballs from git repos
* granular exit codes

Comment (boegel) [reply docs/easybuild-v5/release-candidates.md:32]: I consider granular exit code too much of a detail here.

What should I mention exactly on `EB_PYTHON_PREFIX`, is that really relevant here?

This is by no means meant to be a complete overview.

I will include an extra bullet on the reproducible tarballs though

Comment (lexming) [reply docs/easybuild-v5/release-candidates.md:32]: The important bit about `EB_PYTHON_PREFIX` is that Python packages provided through EasyBuild can now be overwritten by user installations with venvs.

File: docs/easybuild-v5/release-candidates.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+### Highlighted changes
+
+EasyBuild v5.0.0 will include some significant changes compared to EasyBuild v4.x that you should be aware of, including:
+
+- **Python >= 3.6** is required to run EasyBuild (Python >= 3.9 is recommended);
+- using **Lmod >= v8.0** or **Environment Modules v4.3.0** is required as modules tool (check with `module --version`);
+- a new function named `run_shell_cmd` is used to run shell commands like `cmake`, `make`, `pip`, etc. (more info [here](run_shell_cmd.md));
+- the **default settting of some configuration options has changed**:
+    - **RPATH linking** is used by default (can be disabled with `--disable-rpath`);
+    - **trace output** is enabled by default (can be disabled with `--disable-trace`);
```
[the remark this answers is not in the document]
Comment (boegel) [reply docs/easybuild-v5/release-candidates.md:27]: Already done for `--module-depends-on`, but will also cover `--module-extensions`

File: docs/easybuild-v5/release-candidates.md (hunk 1 above)
Comment (boegel) [reply docs/easybuild-v5/release-candidates.md:40]: That seems difficult to explain in a one-liner to me, we'll need a dedicated section in the documentation on that?

**Review by jfgrimm — APPROVED** [review]
lgtm

**Pull request merged**